src/main/java/org/spongepowered/common/data/processor/data/DyeableDataProcessor.java

```diff
@@ -79,7 +82,7 @@ protected DyeableData createManipulator() {
     @Override
     public boolean supports(DataHolder dataHolder) {
         return dataHolder instanceof EntityWolf || dataHolder instanceof EntitySheep || dataHolder instanceof ItemStack &&
-                isDyeable(((ItemStack) dataHolder).getItem());
+                isDyeable(((ItemStack) dataHolder).getItem()) || dataHolder instanceof TileEntityShulkerBox;
```

Comment:
Yes... That's the whole point of the data processors. If anything, multiple processors focusing on doing "one thing well" is part of what the processors are written for, and why there's multiple value processors.